packages/EasyCore/tests/Bridge/Symfony/Serializer/TrimStringsDenormalizerTest.php

```diff
@@ -0,0 +1,200 @@
+            'context has not `TRIM_STRINGS_ALREADY_CALLED` key' => [
+                'data' => 'some-correct-value',
+                'type' => 'no-matter',
+                'format' => 'no-matter',
+                'context' => [
+                    'some-key' => 'some-value',
+                ],
+            ],
+            'data is an array' => [
+                'data' => [],
+                'type' => 'no-matter',
+                'format' => 'no-matter',
+                'context' => null,
+            ],
+            'data is a string' => [
+                'data' => 'some-correct-value',
+                'type' => 'no-matter',
+                'format' => 'no-matter',
+                'context' => null,
+            ],
+        ];
+    }
+
+    /**
+     * @return mixed[]
+     *
+     * @see testDenormalizeSucceedsWithTrimValue
+     */
+    public function provideDataForDenormalize(): array
+    {
+        return [
+            'data is string' => ['', ''],
+            'data is array' => [[], []],
+        ];
+    }
+
+    /**
+     * @return mixed[]
+     *
+     * @see testSupportsDenormalizationReturnsFalse
+     */
+    public function provideIneligibleDataForDenormalization(): array
+    {
+        return [
+            'already called' => [
+                'data' => 'some-correct-value',
+                'type' => 'no-matter',
+                'format' => 'no-matter',
+                'context' => [
+                    'TRIM_STRINGS_ALREADY_CALLED' => true,
+                ],
+            ],
+            'data is an object' => [
+                'data' => new stdClass(),
+                'type' => 'no-matter',
+                'format' => 'no-matter',
+                'context' => null,
+            ],
+            'data is an integer' => [
+                'data' => 123,
+                'type' => 'no-matter',
+                'format' => 'no-matter',
+                'context' => null,
+            ],
+            'data is a float' => [
+                'data' => 12.34,
+                'type' => 'no-matter',
+                'format' => 'no-matter',
+                'context' => null,
+            ],
+            'data is a bool' => [
+                'data' => true,
+                'type' => 'no-matter',
+                'format' => 'no-matter',
+                'context' => null,
+            ],
+        ];
+    }
+
+    /**
+     * @param mixed $data
+     * @param mixed $expectedResult
+     *
+     * @dataProvider provideDataForDenormalize
+     *
+     * @throws \Symfony\Component\Serializer\Exception\ExceptionInterface
+     */
+    public function testDenormalizeSucceedsWithTrimValue($data, $expectedResult): void
+    {
+        $type = 'no-matter';
+        $format = null;
+        $context = [];
+        $except = ['some-key'];
+        $expectation = static function (MockInterface $mock): void {
+            $mock->shouldNotReceive('trim');
+        };
+        if (\is_string($data) || \is_array($data)) {
```

Comment:
This check is not needed, because the data provider contain only a string and an array.

Reply:
Done.